src/dict.c

```diff
@@ -1474,10 +1474,10 @@ static void _dictShrinkIfNeeded(dict *d)
     /* If we reached below 1:10 elements/buckets ratio, and we are allowed to resize
      * the hash table (global setting) or we should avoid it but the ratio is below 1:50,
      * we'll trigger a resize of the hash table. */
-    if ((dict_can_resize == DICT_RESIZE_ENABLE &&  
-         d->ht_used[0] * 100 / DICTHT_SIZE(d->ht_size_exp[0]) < HASHTABLE_MIN_FILL) ||
+    if ((dict_can_resize == DICT_RESIZE_ENABLE &&
+         (double)d->ht_used[0] / DICTHT_SIZE(d->ht_size_exp[0]) <= HASHTABLE_MIN_FILL) ||
         (dict_can_resize != DICT_RESIZE_FORBID &&
-         d->ht_used[0] * 100 / DICTHT_SIZE(d->ht_size_exp[0]) < HASHTABLE_MIN_FILL / dict_force_resize_ratio))
+         (double)d->ht_used[0] / DICTHT_SIZE(d->ht_size_exp[0]) <= HASHTABLE_MIN_FILL / dict_force_resize_ratio))
```

Comment:
In the previous version, 10% is a safe threshold and won't trigger a shrinking. I think we should keep this to keep consistency with `htNeedsShrink`.

Reply:
i think htNeedsShrink also need to do the same change, but it is hard to say which one is right.

the other one is true, but i don't want to rely on the assumption of 2 power, i think readability and simplicity are more important here